pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -415,7 +415,7 @@ func (s *Controller) convertWorkloadEntryToWorkloadInstance(cfg config.Config, c
 		// k8s can't use uds for service objects
 		dnsServiceEntryOnly = true
 	}
-	if !netutil.IsValidIPAddress(addr) {
+	if addr != "" && !netutil.IsValidIPAddress(addr) {
```

Comment:
We add to workloadIndex keyed by address,  i can not find it was done with empty address now

Reply:
Added a guard in workloadindex. The only time we query the index by IP is for GetProxyServiceInstances which shouldn't have a "" ip anyway.

Reply:
Then now if a proxy defined as a WLE with address unset but only with network, would  `GetProxyServiceInstances` return nil.

Then we could not get the service related to this proxy?

Reply:
The proxy for this workload wouldn't connect to this control plane.

Reply:
Is this by design?

Reply:
Yes. This is intended for discovering proxies that are on another network without the control plane having access to the API server of where the proxies reside.

Reply:
If the proxy can connect to the control plane, then we would use a workload entry that has an address.